app/src/battery.c

```diff
@@ -105,4 +107,26 @@ static int zmk_battery_init(const struct device *_arg) {
     return 0;
 }
 
+static int battery_event_listener(const zmk_event_t *eh) {
+
+    if (as_zmk_activity_state_changed(eh)) {
+        switch (zmk_activity_get_state()) {
+        case ZMK_ACTIVITY_ACTIVE:
+            k_timer_start(&battery_timer, K_NO_WAIT, K_SECONDS(CONFIG_ZMK_BATTERY_REPORT_INTERVAL));
```

Comment:
You should pull this and the duplicates line from the init function into a small local `start_reporting` function.